src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -83,12 +83,14 @@ static void HandleNodeIdResolve(void * context, DnssdService * result, CHIP_ERRO
     {
         proxy->OnNodeIdResolutionFailed(PeerId(), error);
         proxy->Release();
+        return;
```

Comment:
Suggest adding a VerifyOrDie below:

```
ResolverDelegateProxy * proxy = static_cast<ResolverDelegateProxy *>(context);

// ADD the below:
VerifyOrDie(proxy != nullptr);
```

The issue is that if somehow proxy is null here, we are corrupted. If we are corrupted, we should not try to call anything, and there will still be a crash on trying to do OnNodeIdResolutionFailed/Release

Reply:
Good catch, I added the VerifyOrDie calls.